### ISSUE TITLE
RLS: 3.0

### ISSUE THREAD

**Issue by datapythonista** [pandas-dev/pandas#57064]
Estimated release date: April 2024, but it will be updated closer to that time.

**Comment by mroeschke** [comment]
Looking like this this release will happen post April 2024. Kicking the tires on things that are left to do before 3.0

Blockers for 3.0.

- [ ] Enforcing deprecations in 2.x https://github.com/pandas-dev/pandas/issues/50578
    - Except (potentially) resampling with `PeriodIndex` https://github.com/pandas-dev/pandas/issues/57033, https://github.com/pandas-dev/pandas/issues/56588
- [x] Doing some action around pyarrow as dependency & the default string type https://github.com/pandas-dev/pandas/issues/57073, https://github.com/pandas-dev/pandas/issues/54466

Notable non-blockers for 3.0

- [x] Drop Python 3.9 support per [NEP 29](https://numpy.org/neps/nep-0029-deprecation_policy.html#support-table) and [SPEC-0](https://scientific-python.org/specs/spec-0000/#support-window)
- [ ] Separate out pandas tests into it's own package https://github.com/pandas-dev/pandas/pull/53007
- [x] Deprecate `make_block` and introduce alternative https://github.com/pandas-dev/pandas/issues/56815, https://github.com/pandas-dev/pandas/pull/58197


Anything else @pandas-dev/pandas-core

**Comment by lithomas1** [comment]
It's probably wise to get in a 2.2.3 release before 3.0.

**Comment by rhshadrach** [comment]
> Anything else **pandas-dev/pandas-core**

A minor issue, but I'd like some eyes on #57351, make sure we're still good with the removal of swapaxes.

**Comment by Aloqeely** [comment]
> Anything else 

If [PDEP-17](https://github.com/pandas-dev/pandas/pull/59125) goes through it'd be great if I could to switch all our deprecations from `FutureWarning` to `DeprecationWarning` before 3.0 — not a blocker though

**Comment by EwoutH** [comment]
> Estimated release date: April 2024

Any update on the estimated release date?

**Comment by BastienGauthier** [comment]
Hello everyone ! I hear everywhere that pandas 3.0 will be awesome, bravo !
Do you have a better view on when we should expect a release (or rc) ? Is it next month/quarter/year ?
Thanks again for the fantastic work done !

**Comment by jbrockmendel** [comment]
Can't speak for everybody, but I think we're getting close and am optimistic about an RC in August or September.

**Comment by timkofu** [comment]
Really looking forward to the free-threading support.

**Comment by mroeschke** [comment]
As of the 2025-10-22 dev meeting, the blockers for a 3.0 RC are:

- [ ] https://github.com/pandas-dev/pandas/issues/58989
- [ ] https://github.com/pandas-dev/pandas/pull/51466 (updating the return values(?) cc @jorisvandenbossche)

**Comment by d-reynol** [comment]
Any timeline for a 3.0 rc?

**Comment by vkingnew** [comment]
when release pandas 3.0 GA？

**Comment by mroeschke** [comment]
Discussed at the dev meeting today, the following PRs should be merged before releasing 3.0rc1 by EOW by @jorisvandenbossche :

- [ ] https://github.com/pandas-dev/pandas/pull/63212
- [ ] https://github.com/pandas-dev/pandas/pull/63018
- [ ] https://github.com/pandas-dev/pandas/pull/63196
- [ ] https://github.com/pandas-dev/pandas/pull/63146
- [ ] https://github.com/pandas-dev/pandas/pull/62552
- [ ] https://github.com/pandas-dev/pandas/pull/63195

**Comment by Elliottt-Chen** [comment]
I noticed the latest nightly wheel (pandas 3.0.0.dev0+2752.ga885d67965) doesn't allow indexing into list elements produced by str.split() when using the pyarrow-backed string dtype.

s1 = pd.Series(...)
s1.str.split().str[...]        # attempting to access list items via str accessor
s1.str.split().list[-1]        # negative index access via list accessor

Specifically, list element access via slice syntax (str[...]) and negative indexing (list[-1]) appear to be unsupported for pyarrow string arrays in the current nightly build.

**Comment by rhshadrach** [comment]
@Elliottt-Chen can you open a new issue.

**Comment by jorisvandenbossche** [comment]
I can cut the RC tomorrow

**Comment by jorisvandenbossche** [comment]
First RC is released, and is on PyPI and also on conda-forge now. 

Thus can be installed with:

```
python -m pip install --upgrade --pre pandas==3.0.0rc0
```
or

```
conda install -c conda-forge/label/pandas_rc pandas==3.0.0rc0
```

Will do an announcement on the mailing list tomorrow (and linkedin/mastodon)

**Comment by notatallshaw** [comment]
FYI, no need to call `--pre` if you're explicitly requesting a pre-release version, the `--pre` will enable pre-releases for all packages, not just pandas: 

```
python -m pip install --upgrade pandas==3.0.0rc0
```

**Comment by tswast** [comment]
How would you like potential regressions, such as https://github.com/pandas-dev/pandas/issues/63304, reported? Should we ping this bug or just file as normally?

**Comment by mroeschke** [comment]
> How would you like potential regressions, such as https://github.com/pandas-dev/pandas/issues/63304, reported? Should we ping this bug or just file as normally?

Filing normally noting that it's a 3.0rc bug (like you did) is preferred, thanks!

**Comment by mroeschke** [comment]
Related thoughts from the 2025-12-10 dev meeting:

* Opened [issues](https://github.com/pandas-dev/pandas/issues?q=is%3Aissue%20state%3Aopen%20milestone%3A3.0) so far during the RC period would be nice to fix but not a blocker for the official release (i.e. able to be fixed in 3.0.x) 
* @jbrockmendel's `Timedelta(int, unit=)` PR's should be merged before an RC2 (if agreed OK as breaking to match datetime behavior on main)
    * https://github.com/pandas-dev/pandas/pull/63302
    * https://github.com/pandas-dev/pandas/pull/63303

**Comment by jorisvandenbossche** [comment]
I was thinking to cut a second RC tomorrow, although all the datetime/timedelta PRs won't be merged by then. But if we do another RC next week that is maybe fine?

**Comment by jbrockmendel** [comment]
Does your plan involve doing a 3rd RC down the line?  bc if so I'd rather wait to get the datetime stuff in so we can keep it to just 2 RCs.

**Comment by jorisvandenbossche** [comment]
> Does your plan involve doing a 3rd RC down the line?

Yes, as I am saying above, we can do another RC next week. It is relatively cheap to do RCs, so I would rather do one more. There are already quite some changes since the first RC of two weeks ago.

**Comment by mroeschke** [comment]
After the 2026-01-14 meeting

1. The last 3.0rc to be released today by @jorisvandenbossche 
2. The remaining blockers for 3.0
    * https://github.com/pandas-dev/pandas/pull/63613
    * https://github.com/pandas-dev/pandas/pull/63473
    * https://github.com/pandas-dev/pandas/pull/63666
3. 3.0 to be released early next week (e.g. Monday, Jan. 19)
    *  Reminder to sync dependency versions when releasing on conda-forge https://github.com/conda-forge/pandas-feedstock/pull/239
    * Make tzdata an optional dependency

**Comment by Dr-Irv** [comment]
Also update the docs page to have a 3.0 pulldown

**Comment by jorisvandenbossche** [comment]
I cut the RC2 yesterday, and it is on PyPI, but running into build issues on conda-forge feedstock: https://github.com/conda-forge/pandas-feedstock/pull/242

**Comment by tswast** [comment]
https://github.com/pandas-dev/pandas/issues/57064#issuecomment-3751175039 Monday is a holiday here in the US (MLK Day). I'm curious if it would be possible to do Tuesday, instead?

**Comment by jorisvandenbossche** [comment]
Sure! (I would only get to it Monday evening the earliest anyway)

But at the moment, releasing is also being blocked by finding a fix for the conda feedstock

**Comment by rhshadrach** [comment]
It'd also be good to get #63740 in.

**Comment by jorisvandenbossche** [comment]
Status update: I opened a few PRs with some more release note tweaks today. I plan to merge the remaining open PRs currently tagged as 3.0 tomorrow morning, and then cut the release (so reviews of those PRs is certainly welcome: https://github.com/pandas-dev/pandas/pulls?q=is%3Aopen+is%3Apr+milestone%3A3.0).

**Comment by jorisvandenbossche** [comment]
The 3.0.0 release is cut (https://github.com/pandas-dev/pandas/releases/tag/v3.0.0) and up on PyPI. 
The build on conda-forge is currently having some issue on windows: https://github.com/conda-forge/pandas-feedstock/pull/243

Announcement blog post can use some review: https://github.com/pandas-dev/pandas/pull/63796 
(planning to send around announcement tomorrow / when conda-forge is ready)

**Comment by khmyznikov** [comment]
Congrats! 🥳

**Comment by mroeschke** [comment]
Closing as completed
